### PR TITLE
Core, Flink, Spark, KafkaConnect: Remove remaining usage of deprecated path API

### PR DESCRIPTION
**Pull request by amogh-jahagirdar**
Same as https://github.com/apache/iceberg/pull/11563, removing the remaining usages of the path API and replacing it with location API

### CODE REVIEW

File: flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java [hunk 1]
```diff
@@ -602,7 +602,7 @@ public static void assertEquals(ContentFile<?> expected, ContentFile<?> actual)
     assertThat(actual).isNotNull();
     assertThat(actual.specId()).as("SpecId").isEqualTo(expected.specId());
     assertThat(actual.content()).as("Content").isEqualTo(expected.content());
-    assertThat(actual.path()).as("Path").isEqualTo(expected.path());
+    assertThat(actual.location()).as("Path").isEqualTo(expected.location());
```
Comment (ebyhr) [inline flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:605]:
```suggestion
    assertThat(actual.location()).as("Location").isEqualTo(expected.location());
```

File: spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java [hunk 2]
```diff
@@ -300,7 +300,7 @@ public void testExpireDeleteFiles() throws Exception {
     Assert.assertEquals("Should have 1 delete file", 1, TestHelpers.deleteFiles(table).size());
     Path deleteManifestPath = new Path(TestHelpers.deleteManifests(table).iterator().next().path());
     DeleteFile deleteFile = TestHelpers.deleteFiles(table).iterator().next();
-    Path deleteFilePath = new Path(String.valueOf(deleteFile.path()));
+    Path deleteFilePath = new Path(String.valueOf(deleteFile.location()));
```
Comment (ebyhr) [inline spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java:303]: Why do we need `String.valueOf` even though `deleteFile.location()` is String?

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ public boolean next() throws IOException {
       if (currentTask != null && !currentTask.isDataTask()) {
         String filePaths =
             referencedFiles(currentTask)
-                .map(file -> file.path().toString())
+                .map(file -> file.location())
```
Comment (ebyhr) [inline spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java:151]: Spark v3.5 uses method reference here. Feel free to ignore this and other comments about method reference if it's intentional.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkCleanupUtil.java [hunk 4]
```diff
@@ -81,7 +81,7 @@ private static String taskInfo() {
    * @param files a list of files to delete
    */
   public static void deleteFiles(String context, FileIO io, List<? extends ContentFile<?>> files) {
-    List<String> paths = Lists.transform(files, file -> file.path().toString());
+    List<String> paths = Lists.transform(files, file -> file.location());
```
Comment (ebyhr) [inline spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkCleanupUtil.java:84]: Spark v3.5 uses method reference here.

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/ValidationHelpers.java [hunk 5]
```diff
@@ -42,7 +42,7 @@ public static List<Long> snapshotIds(Long... ids) {
   }
 
   public static List<String> files(ContentFile<?>... files) {
-    return Arrays.stream(files).map(file -> file.path().toString()).collect(Collectors.toList());
+    return Arrays.stream(files).map(file -> file.location()).collect(Collectors.toList());
```
Comment (ebyhr) [inline spark/v3.3/spark/src/test/java/org/apache/iceberg/ValidationHelpers.java:45]: nit: We could use method reference. Feel free to ignore this comment. I know Spark v3.5 uses this style.

File: spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java [hunk 6]
```diff
@@ -635,7 +635,7 @@ private static void importUnpartitionedSparkTable(
       if (checkDuplicateFiles) {
         Dataset<Row> importedFiles =
             spark
-                .createDataset(Lists.transform(files, f -> f.path().toString()), Encoders.STRING())
+                .createDataset(Lists.transform(files, f -> f.location()), Encoders.STRING())
```
Comment (ebyhr) [inline spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java:638]: Spark v3.5 uses method reference here.

File: spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java [hunk 7]
```diff
@@ -153,7 +153,7 @@ public boolean next() throws IOException {
       if (currentTask != null && !currentTask.isDataTask()) {
         String filePaths =
             referencedFiles(currentTask)
-                .map(file -> file.path().toString())
+                .map(file -> file.location())
```
Comment (ebyhr) [inline spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java:156]: Spark v3.5 uses method reference here.

File: spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkCleanupUtil.java [hunk 8]
```diff
@@ -81,7 +81,7 @@ private static String taskInfo() {
    * @param files a list of files to delete
    */
   public static void deleteFiles(String context, FileIO io, List<? extends ContentFile<?>> files) {
-    List<String> paths = Lists.transform(files, file -> file.path().toString());
+    List<String> paths = Lists.transform(files, file -> file.location());
```
Comment (ebyhr) [inline spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkCleanupUtil.java:84]: Spark v3.5 uses method reference here.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java (hunk 3 above)
Comment (amogh-jahagirdar) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java:151]: Let's keep it consistent, I'll fix it!

File: spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java (hunk 2 above)
Comment (amogh-jahagirdar) [reply spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java:303]: We don't, good catch!

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java [hunk 9]
```diff
@@ -867,18 +867,17 @@ private List<DeleteFile> deleteFiles(Table table) {
...
     return first.stream()
-        .filter(f -> !secondPaths.contains(f.path().toString()))
+        .filter(f -> !secondPaths.contains(f.location()))
         .collect(Collectors.toList());
   }
 
   private void assertNotContains(List<DeleteFile> original, List<DeleteFile> rewritten) {
     Set<String> originalPaths =
-        original.stream().map(f -> f.path().toString()).collect(Collectors.toSet());
+        original.stream().map(f -> f.location()).collect(Collectors.toSet());
```
Comment (nastra) [inline spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:878]: nit: there are a bunch of places like this which can be updated to `.map(ContentFile::location)`

**Review by nastra — APPROVED** [review]

**Comment by amogh-jahagirdar** [comment]
Thanks for reviewing @nastra @ebyhr!

**Pull request merged**